### PR TITLE
Click outside midpoint

### PR DESCRIPTION
**Pull request by huwshimi**
Hide the midpoint when clicking outside it.

### CODE REVIEW

**Comment by huwshimi** [comment]
QA:
- click in the search box
- the mid-point should open
- click outside the mid-point
- it should close

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/3105/
Test PASSed.

File: npm-shrinkwrap.json [hunk 1]
```diff
@@ -2567,39 +2567,39 @@
     },
     "handlebars": {
       "version": "4.0.2",
-      "from": "handlebars@",
+      "from": "https://registry.npmjs.org/handlebars/-/handlebars-4.0.2.tgz",
```
Comment (huwshimi) [inline npm-shrinkwrap.json]: Not sure why shrinkwrap made all these changes. Is this OK?

**Comment by urosj** [comment]
:+1: on QA, does not work in Safari (works in Chrome & FF)

File: jujugui/static/gui/src/app/components/header-search/header-search.js [hunk 2]
```diff
@@ -47,6 +47,19 @@ YUI.add('header-search', function() {
     },
 
     /**
+      Update the state when the app state changes.
+
+      @method componentWillReceiveProps
+    */
+    componentWillReceiveProps: function() {
+      var component = this.props.getAppState(
```
Comment (hatched) [inline jujugui/static/gui/src/app/components/header-search/header-search.js]: Why is this necessary? The empty method should be being called to unmount this if there is no component defined so it shouldn't have to close itself.

File: npm-shrinkwrap.json (hunk 1 above)
Comment (hatched) [inline npm-shrinkwrap.json]: Yeah I'm not sure why it does this either.

**Comment by hatched** [comment]
:+1: QA :ok:

File: jujugui/static/gui/src/app/components/header-search/header-search.js (hunk 2 above)
Comment (hatched) [inline jujugui/static/gui/src/app/components/header-search/header-search.js]: After qa'ing I see why this is here. Could you maybe add a comment :) thanks

File: npm-shrinkwrap.json (hunk 1 above)
Comment (huwshimi) [inline npm-shrinkwrap.json]: Is it OK to leave these in?

File: jujugui/static/gui/src/app/components/header-search/header-search.js (hunk 2 above)
Comment (huwshimi) [inline jujugui/static/gui/src/app/components/header-search/header-search.js]: Will do.

File: npm-shrinkwrap.json (hunk 1 above)
Comment (hatched) [inline npm-shrinkwrap.json]: yup

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/3121/
Test PASSed.

**Comment by huwshimi** [comment]
@urosj I've added a card to fix this in safari, just so we can keep moving at the moment.

Thanks for the reviews :shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge
